### PR TITLE
New mpl drawer fixes open issues and adds proportional font text width spacing and labels

### LINKED ISSUE #3641: Drawing initialize disordered in mpl drawer
### Information

- **Qiskit Terra version**: 0.12.0.dev0+6b6170a
- **Python version**: 3.6
- **Operating system**: Ubuntu 18.04

### What is the current behavior?
The box of 'initialize' is disordered. It gets outside of the circuit and overlaps with other gates.

initialize-error

### Steps to reproduce the problem

```from qiskit import *
from math import pi, sqrt
import numpy as np

vec = np.array([sqrt(1/2), 0, 0, sqrt(1/2)])

qc = QuantumCircuit(2)
qc.initialize(vec, [0,1])
qc.u3(0.12,pi/2,pi/4,0)
qc.draw(output="mpl")
```

### LINKED ISSUE #3766: MPL drawer should prioritize instruction label over name
# Information

- **Qiskit Terra version**: master
- **Python version**: 3.7
- **Operating system**: MacOS

### What is the current behavior?

When an instruction has a label, the label should be used. If a no-boxing instruction has a label, a box should be used.  

### Steps to reproduce the problem
```python
from qiskit import QuantumRegister, QuantumCircuit
from qiskit.extensions import XGate, UnitaryGate, CzGate
import numpy

qr = QuantumRegister(2, 'q')
circ = QuantumCircuit(qr)
circ.append(XGate(), [qr[0]])
circ.append(XGate(label='alt-X'), [qr[0]])
circ.append(UnitaryGate(numpy.eye(4)), [qr[0], qr[1]])
circ.append(UnitaryGate(numpy.eye(4), label="iswap"), [qr[0], qr[1]])
circ.append(CzGate(), [qr[0], qr[1]])
circ.append(CzGate(label='cz label'), [qr[0], qr[1]])
circ.draw('mpl')
```
image

### What is the expected behavior?

Something like the text drawer:
```
        ┌───┐┌───────┐┌──────────┐┌────────┐   ┌───────────┐
q_0: |0>┤ X ├┤ alt-X ├┤0         ├┤0       ├─■─┤0          ├
        └───┘└───────┘│  Unitary ││  iswap │ │ │  cz label │
q_1: |0>──────────────┤1         ├┤1       ├─■─┤1          ├
                      └──────────┘└────────┘   └───────────┘
```

### LINKED ISSUE #4148: Circuit marplot lib register and measurement on-top of each other
### What is the expected enhancement?
If the circuit goes over two frames the second frame if it is a measurement the register and the value are on top of each other -- see figure. 

image

### LINKED ISSUE #4248: Some open-controlled gates incorrectly drawn in mpl drawer
### Information

- **Qiskit Terra version**: 0.14.0.dev0+d0a487b
- **Python version**: 3.6
- **Operating system**: Ubuntu 18.04

### What is the current behavior?

Visualization open controlled gates was added in #4131 . But a few gates were missed out. In the circuit below, all were implemented as open-controlled gates but were drawn as below:

err_circuit

### Steps to reproduce the problem

```
qc = QuantumCircuit(3)
qc.append(YGate().control(ctrl_state='0'),[0,1])
qc.append(ZGate().control(ctrl_state='0'),[0,1])
qc.append(HGate().control(ctrl_state='0'),[0,1])
qc.append(U1Gate(pi/3).control(ctrl_state='0'),[0,1])
qc.append(RZGate(pi/4).control(ctrl_state='0'),[0,1])
qc.append(SwapGate().control(ctrl_state='0'),[0,1,2])
                    
qc.draw('mpl')
```

### LINKED ISSUE #4290: MPL drawer should expose cregbundle as a parameter
In the spirit of https://github.com/Qiskit/qiskit-terra/issues/2797, PR https://github.com/Qiskit/qiskit-terra/pull/4274 exposes `cregbundle` as a parameter for circuit drawing.

MPL should use that parameter and deprecate the style dict key.

### LINKED ISSUE #4293: initial_state=True is not honored by mpl drawer
When mpl drawer is called with `initial_state=True` the initial values are not there:
```
from qiskit import *
qr = QuantumRegister(3, 'q')
cr = ClassicalRegister(3, 'c')
circuit = QuantumCircuit(qr, cr)
circuit.measure(qr, cr)
circuit.draw('mpl', initial_state=True)
```
image

The output should be something like this:

circuit

### LINKED ISSUE #4352: MPL drawer breaks with Isometry
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.7
- **Operating system**: MacOS

When trying to MPL graph an isometry gate, I'm getting a lot of funny outputs, depending on the isometry size.

### What is the current behavior?

considering the following code:
```python
from qiskit import QuantumCircuit
import numpy as np

size =  # <- input
qc = QuantumCircuit(size)
qc.isometry(np.eye(2**size, 2**size), [i for i in range(size)], [])
qc.draw('mpl')
```
```
size = 1
```
image

```
size = 2
```
image
```
size = 3
```
image

### What is the expected behavior?

Something like the text drawer:
```
     ┌──────────┐
q_0: ┤ ISOMETRY ├
     └──────────┘
```

```
     ┌───────────┐
q_0: ┤0          ├
     │  ISOMETRY │
q_1: ┤1          ├
     └───────────┘
```

```
     ┌───────────┐
q_0: ┤0          ├
     │           │
q_1: ┤1 ISOMETRY ├
     │           │
q_2: ┤2          ├
     └───────────┘
```

### LINKED ISSUE #4439: AttributeError when drawing circuit using Matplotlib
### Information

- **Qiskit Terra version**: 0.14.1
- **Python version**: 3.7.6
- **Operating system**: Ubuntu 16.04

### What is the current behavior?

When trying to draw a `QuantumCircuit` object using `.draw(output='mpl')`, I'm getting `AttributeError: 'Instruction' object has no attribute 'label'` in `qiskit/visualization/matplotlib.py`. The circuit draws fine using the default text drawer.

### Steps to reproduce the problem

The code is adapted from the `linear_systems_of_equations` [HHL tutorial notebook](https://github.com/Qiskit/qiskit-tutorials/blob/master/legacy_tutorials/aqua/linear_systems_of_equations.ipynb). Here's a (probably non-minimal) snippet to reproduce:
```
from qiskit.aqua.algorithms import HHL
from qiskit.aqua.components.eigs import EigsQPE
from qiskit.circuit.library import QFT
from qiskit.aqua.components.reciprocals import LookupRotation
from qiskit.aqua.operators import MatrixOperator
from qiskit.aqua.components.initial_states import Custom

def create_eigs(matrix, num_ancillae, negative_evals):
    ne_qfts = [None, None]
    if negative_evals:
        num_ancillae += 1
        ne_qfts = [QFT(num_ancillae - 1), QFT(num_ancillae - 1, inverse=True)]

    return EigsQPE(MatrixOperator(matrix=matrix),
                   QFT(num_ancillae, inverse=True),
                   num_time_slices=50,
                   num_ancillae=num_ancillae,
                   expansion_mode='suzuki',
                   expansion_order=2,
                   evo_time=None,
                   negative_evals=negative_evals,
                   ne_qfts=ne_qfts)

matrix = [[1, 0], [0, 2]]
vector = [1, 4]

orig_size = len(vector)
matrix, vector, truncate_powerdim, truncate_hermitian = HHL.matrix_resize(matrix, vector)

# Initialize eigenvalue finding module
eigs = create_eigs(matrix, 3, False)
num_q, num_a = eigs.get_register_sizes()

# Initialize initial state module
init_state = Custom(num_q, state_vector=vector)

# Initialize reciprocal rotation module
reci = LookupRotation(negative_evals=eigs._negative_evals, evo_time=eigs._evo_time)

algo = HHL(matrix, vector, truncate_powerdim, truncate_hermitian, eigs,
           init_state, reci, num_q, num_a, orig_size)

circuit = algo.construct_circuit(measurement=True)
circuit.draw(output='mpl')
```

### What is the expected behavior?

No errors, and the circuit is drawn correctly using Matplotlib.

### Suggested solutions
Not sure - I don't know whether this could be a Terra visualization problem or an Aqua circuit design problem (or just a problem with the code above).

### LINKED ISSUE #4449: MPL drawer leaves lots of white space to left of transpiled circuits
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
Original:
Screen Shot 2020-05-14 at 09 18 24

Transpiled:

Screen Shot 2020-05-14 at 09 18 38

### LINKED ISSUE #4496: Generic Gates have no color in MPL circuit drawer
From [Qiskit Textbook 2.5](https://qiskit.org/textbook/ch-gates/more-circuit-identities.html):

```python

A = Gate('A', 1, [])
B = Gate('B', 1, [])
C = Gate('C', 1, [])

qc = QuantumCircuit(2)
qc.append(C, [t])
qc.cz(0,1)
qc.append(B, [t])
qc.cz(0,1)
qc.append(A, [t])
qc.draw('mpl')
```

image

### LINKED ISSUE #4497: mpl drawer should use text width instead of text len
### What is the expected enhancement?
The mpl drawer uses the number of chars in a name, label, etc. in order to determine its width. This works fine for the text drawer, but since the mpl drawer uses a proportional font, this leads to inconsistent drawing. The text len is used to determine the box size for longer names and the layer width. It is also used to get the width of the params. For this code,
```
from qiskit import QuantumRegister, QuantumCircuit, ClassicalRegister

qr = QuantumRegister(8, 'thisisanarrowiiiiiiiiiiiiname')
qc = ClassicalRegister(8, 'tHISISAWIDEWWWWWWWWWWWWNAME')
ct = QuantumCircuit(qr, qc, name="My Circuit")
g_circuit = QuantumCircuit(3, name='thisisanarrowiiiiiiiiiiiiname')
g_circuit.h(0)
g_circuit.cx(0, 1)
g = g_circuit.to_gate()
ct.append(g, [0,1,2])
g_circuit2 = QuantumCircuit(3, name='tHISISAWIDEWWWWWWWWWWWWNAME')
g_circuit2.h(0)
g_circuit2.cx(1, 2)
g = g_circuit2.to_gate()
ct.append(g, [4, 5, 6]
ct.draw(output='mpl', filename='testtogate.png')
```
The current master produces this output,
image
Notice that the gate with the narrow name is actually wider than the one with a wide name because there are more chars in the narrow name. By using a text width calculation, it is possible to get the following,
image

### LINKED ISSUE #4506: circuit draw() does not show properly when you use output='mpl' and fold=-1
### Information

- **Qiskit Terra version**: 0.14.1
- **Python version**: 3.7.4
- **Operating system**: Darwin

### What is the current behavior?
For a quantum circuit, qc, when you run qc.draw(output='mpl', fold=-1) the circuit is drawn in one line but the horizontal lines and registers are gone. Works with output='text'.

### Steps to reproduce the problem
q = QuantumRegister(2, 'q')
c = ClassicalRegister(2, 'c')
qc = QuantumCircuit(q,c)

for _ in range(40):
    qc.h(q[0])
    qc.h(q[1])
qc.measure(q,c)
qc.draw(output='mpl',fold=-1)

### What is the expected behavior?
Should have proper figure

### LINKED ISSUE #4545: Controlled UnitaryGate not displayed correctly.
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
This issue was brought up in the Qiskit Slack workspace. 
https://qiskit.slack.com/archives/C7SJ0PJ5A/p1591208099333800?thread_ts=1591204734.332900&cid=C7SJ0PJ5A

### Steps to reproduce the problem
```
from qiskit import *
from qiskit.extensions import UnitaryGate
import numpy as np
c=QuantumCircuit(3)
op = np.matrix([[1,0],[0,-1]])
r=UnitaryGate(op, label='rt')
re=r.control(2, ctrl_state='11')
c.append(re, [0,1,2])
c.draw('mpl')
```

### LINKED ISSUE #4580: Labels should be added to mpl drawer
### What is the expected enhancement?
Labels were added to text drawer in #4361, #4262, and #4420. Labels are user-supplied text that overwrites the names of gates and can be added to controls. This needs to be implemented for the mpl drawer. This issue is intended as a discussion to come to a consensus on design before we start coding.

image
The control labels can be placed on the highest control or below the lowest control if there are no controls above. And placed inside or outside the control line. In addition to what's shown here, some other possible questions would be,
1. If there are no params, should the original gate name be placed in the gate box underneath the gate label in smaller text? Might help prevent confusion if someone did `YGate(label='X')`
2. Or should the original gate name be placed in parens at the end of the label? This would work whether there are params or not and since most gate names are short would not take up too much space in most cases.
3. What do we do with gate labels for swap, cz, cu1, rzz, cx's and possibly others that don't have a box to display in? Should we use sidetext for these?
4. Currently all gate names are capitalized. Should all labels be capitalized or left as the user enters them?

One other thing. `qc.cy(0, 1, label='Y Label')` will always produce a control label. In the current implementation, the data structure is the same as 
`qc.append(YGate().control(1, label='Y Label'), [0, 1])`.
 If you want a gate label, you have to use a YGate form.

### LINKED ISSUE #4605: Bad representation of control rotation gates
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behaviour?

There are situations where matplotlib representation of the circuit fails

### Steps to reproduce the problem

In the following circuit example, crx gates and x gates are overlapped, wires are not shown so the entire circuit is not well represented by matplotlib however the ASCII representation is correct:

image

### What is the expected behaviour?

A correct visualisation of the circuit using mpl

### LINKED ISSUE #4611: U gates with long parameters generate a poor visualization
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?

U gates with long parameters generate a poor visualization using matplotlib as can be seen in the following example that compares ASCII and mpl:

image
### Steps to reproduce the problem

from qiskit.circuit.library.data_preparation import PauliFeatureMap

prep = PauliFeatureMap(2, reps=1, paulis=['ZZ'])
prep.draw('mpl')

### What is the expected behavior?

a clear and more simple visualization of U gates parameters

### Suggested solutions

option: remove operations inside the parameter and calculate the value
option: short the length of decimal values
option: do gate pink block much bigger to accommodate the string

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

This new version of the mpl drawer fixes most of the open mpl issues, adds true text width spacing for the proportional font, adds the display of gate and control labels, and restructures the matplotlib.py code.

### Details and comments

Below are 2 images of a test circuit. The first is generated by the current master (v1) and the second by the new version (v2). The issues on the v1 image are as follows. All are fixed on v2.
- The col index numbers at the top do not line up with the gates and do not shift over when a longer register name is used.
- The wide register name runs off the page to the left on v1. 
- For short register names, such as q0 (not shown here), v1 leaves excessive space to the left of the name. This is removed in v2, providing more space for displaying the circuit.
- There are no gate or control labels displayed in v1.
- There is inconsistent capitalization of gate names in v1.
- The U2 gate in the first col and the U3 in the second col are excessively wide for the params shown. (Compare to the U3 in third col.)
- The iswap box is too wide and the name in the ms box runs into the 1.
- The controlled-swap at the end of the first row should not have a box, but the double-x's of a swap.
- The iqp gate is way too wide for the name. It takes up 9 indexes in v1 and 5 in v2.
- The rzz near the right side of row 2 should be a sidetext 'zz' gate instead of a box.
- The Fred... gate created by a to_gate command, overwrites 2 gates and runs off the right edge.
- The controlled-X gates below index 61 should have the target-cross instead of an X box.
- The Hamiltonian creates a great deal of empty space due to the parameters containing an array.
- The Unitary gate should not have a 0 in front of the name since it is a 1-qubit gate.
- The A gate at the end should display in purple.
- isometry and initialize gates are not shown on v1 since they will crash the drawer.
- The background color change does not work on v1.
### Current Master (v1)
image
### New Version (v2)
image
### Other Changes
- The code was restructured and redundancies, inefficiencies, and unused code were removed. Even with adding the text width calculations and the gate and control label displays, the new version of matplotlib.py is 140 lines shorter than the master.
- Several open PR's were incorporated here, including,
      - PR #4396 which adds the cregbundle parameter to the draw command
      - PR #4377 which adds the initial_state parameter to the draw command
      - PR #4518 which fixes a problem with fold=-1.
- Using text len to determine layer and box widths, as is done in the text drawer and in v1, creates very inconsistent width calculations for a proportional font, since it depends on what characters are in the text. The new version now uses a true proportional font text width calculation to determine box and layer widths.
- Gate and control labels were previously added to the text drawer and are now implemented for mpl. Gate labels replace the gate name and control labels are printed above the visual top of the highest control, unless the gate box is above all the controls, in which case the control label will be below the lowest control.

### Fixed Issue Numbers
The following open issues are fixed by this PR,
Fixes #4611
Fixes #4605 
Fixes #4580 
Fixes #4545 
Fixes #4506 
Fixes #4497 
Fixes #4496 
Fixes #4449 
Fixes #4439 
Fixes #4352 
Fixes #4293 
Fixes #4290 
Fixes #4248 
Fixes #4148 
Fixes #3766
Fixes #3641 
### To Do
A reno for cregbundle, initial_state, and labels needs to be done.

### CODE REVIEW

File: qiskit/circuit/quantumcircuit.py [hunk 1]
```diff
@@ -940,7 +940,7 @@ def qasm(self, formatted=False, filename=None):
     def draw(self, output=None, scale=0.7, filename=None, style=None,
              interactive=False, plot_barriers=True,
              reverse_bits=False, justify=None, vertical_compression='medium', idle_wires=True,
-             with_layout=True, fold=None, ax=None, initial_state=False, cregbundle=False):
+             with_layout=True, fold=None, ax=None, initial_state=False, cregbundle=True):
```
Comment (1ucian0) [inline qiskit/circuit/quantumcircuit.py:943]: We are changing the default later, with https://github.com/Qiskit/qiskit-terra/pull/4595, which is blocked because a Latex drawer issue.

File: qiskit/circuit/quantumcircuit.py [hunk 2]
```diff
@@ -1004,10 +1004,9 @@ def draw(self, output=None, scale=0.7, filename=None, style=None,
                 only used when the ``output`` kwarg is set to use the ``mpl``
                 backend. It will be silently ignored with all other outputs.
             initial_state (bool): Optional. Adds ``|0>`` in the beginning of the wire.
-                Only used by the ``text``, ``latex`` and ``latex_source`` outputs.
                 Default: ``False``.
-            cregbundle (bool): Optional. If set True bundle classical registers. Not used by
-                the ``matplotlib`` output. Default: ``False``.
+            cregbundle (bool): Optional. If set True bundle classical registers.
+                Default: ``True``.
```
Comment (1ucian0) [inline qiskit/circuit/quantumcircuit.py:1009]: [same](https://github.com/Qiskit/qiskit-terra/pull/4616/files#r448690073)

File: qiskit/visualization/circuit_visualization.py [hunk 3]
```diff
@@ -65,7 +65,7 @@ def circuit_drawer(circuit,
                    fold=None,
                    ax=None,
                    initial_state=False,
-                   cregbundle=False):
+                   cregbundle=True):
```
Comment (1ucian0) [inline qiskit/visualization/circuit_visualization.py:68]: [same](https://github.com/Qiskit/qiskit-terra/pull/4616/files#r448690073)

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py [hunk 4]
```diff
@@ -114,7 +124,8 @@ def test_plot_barriers(self):
 
         # check for barriers
         circuit.h(q[0])
-        circuit.barrier()
+        circuit.barrier(0)
+        circuit.h(q[0])
```
Comment (1ucian0) [inline test/ipynb/mpl/test_circuit_matplotlib_drawer.py:128]: Why this change? If the new situation is particularly interesting, I would add a new test.

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py [hunk 5]
```diff
@@ -17,11 +17,21 @@
...
+                                    MCXGate, YGate, HGate, ZGate,
+                                    C4XGate, C3XGate, MSGate, RZZGate)
+from qiskit.extensions import HamiltonianGate, UnitaryGate
+from qiskit.circuit import Gate, Parameter
+from qiskit.circuit.library import IQP
 
+import math
+import numpy as np
+
+pi = np.pi
```
Comment (1ucian0) [inline test/ipynb/mpl/test_circuit_matplotlib_drawer.py:34]:
```suggestion
from numpy import pi
```

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py (hunk 4 above)
Comment (enavarro51) [reply test/ipynb/mpl/test_circuit_matplotlib_drawer.py:128]: Just thought since there was a recent fix to displaying partial barriers, it was good to have one. Could be split out to another test, course.

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py [hunk 6]
```diff
@@ -138,6 +149,165 @@ def test_no_barriers_false(self):
...
+        """Test different control-x gates"""
+        qr = QuantumRegister(5, 'q')
+        circuit = QuantumCircuit(qr)
+        circuit.x(0)
+        circuit.cx(0, 1)
+        circuit.ccx(0, 1, 2)
+        circuit.append(XGate().control(3, ctrl_state='010'), [qr[2], qr[3], qr[0], qr[1]])
+        circuit.append(MCXGate(num_ctrl_qubits=3, ctrl_state='101'), [qr[0], qr[1], qr[2], qr[4]])
+
+        self.circuit_drawer(circuit, filename='ctrl_x.png')
```
Comment (1ucian0) [inline test/ipynb/mpl/test_circuit_matplotlib_drawer.py:211]: `control-x` is not very common name for `cx`. I would suggest `cnot`
```suggestion
    def test_cnot(self):
        """Test different cnot gates"""
        qr = QuantumRegister(5, 'q')
        circuit = QuantumCircuit(qr)
        circuit.x(0)
        circuit.cx(0, 1)
        circuit.ccx(0, 1, 2)
        circuit.append(XGate().control(3, ctrl_state='010'), [qr[2], qr[3], qr[0], qr[1]])
        circuit.append(MCXGate(num_ctrl_qubits=3, ctrl_state='101'), [qr[0], qr[1], qr[2], qr[4]])

        self.circuit_drawer(circuit, filename='cnot.png')
```

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py [hunk 7]
```diff
@@ -138,6 +149,165 @@ def test_no_barriers_false(self):
...
+        circuit.x(0)
+        circuit.h(0)
+        circuit.x(1)
+
+        self.circuit_drawer(circuit, filename='creg_initial_true.png',
+                            cregbundle=True, initial_state=True)
+        self.circuit_drawer(circuit, filename='creg_initial_false.png',
+                            cregbundle=False, initial_state=False)
+
+    def test_4_gates(self):
```
Comment (1ucian0) [inline test/ipynb/mpl/test_circuit_matplotlib_drawer.py:260]:
```suggestion
    def test_r_gates(self):
```

File: qiskit/circuit/quantumcircuit.py (hunk 1 above)
Comment (enavarro51) [reply qiskit/circuit/quantumcircuit.py:943]: Just a final word on this. cregbundle True has been the default for mpl, and I think users would be surprised if it became False. Assuming the default gets returned to True soon, it would not be an issue.

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py (hunk 5 above)
Comment (enavarro51) [reply test/ipynb/mpl/test_circuit_matplotlib_drawer.py:34]: Of course.

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py (hunk 6 above)
Comment (enavarro51) [reply test/ipynb/mpl/test_circuit_matplotlib_drawer.py:211]: Either/or. I always assumed a cx was a cnot, a ccx was a Toffoli, and all the others were multi-controlled x's.

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py (hunk 7 above)
Comment (enavarro51) [reply test/ipynb/mpl/test_circuit_matplotlib_drawer.py:260]: Yes. typo.

File: qiskit/circuit/quantumcircuit.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/circuit/quantumcircuit.py:943]: Yes, the goal is to turn it True asap

File: qiskit/visualization/matplotlib.py [hunk 8]
```diff
@@ -35,15 +33,19 @@
...
 
 from qiskit.circuit import ControlledGate
-from qiskit.visualization import exceptions
 from qiskit.visualization.qcstyle import DefaultStyle, BWStyle
 from qiskit import user_config
 from qiskit.circuit.tools.pi_check import pi_check
 
 logger = logging.getLogger(__name__)
+# import matplotlib
+# matplotlib.use('ps') # for testing different backends
```
Comment (1ucian0) [inline qiskit/visualization/matplotlib.py:42]:
```suggestion
```

Comment (enavarro51) [reply qiskit/visualization/matplotlib.py:42]: ok.

**Comment by enavarro51** [comment]
@1ucian0 Just a reminder. After #4616 is merged, you can close PRs #4264, #4377, #4396, #4518, and #4519.

File: qiskit/visualization/circuit_visualization.py (hunk 3 above)
Comment (1ucian0) [reply qiskit/visualization/circuit_visualization.py]: Undone in 95416cb49

File: qiskit/circuit/quantumcircuit.py (hunk 2 above)
Comment (1ucian0) [reply qiskit/circuit/quantumcircuit.py]: undone in 95416cb49

File: qiskit/circuit/quantumcircuit.py (hunk 1 above)
Comment (1ucian0) [reply qiskit/circuit/quantumcircuit.py]: undone in 95416cb49

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py (hunk 6 above)
Comment (1ucian0) [reply test/ipynb/mpl/test_circuit_matplotlib_drawer.py]: Changed to cnot in 750d066b9

File: test/ipynb/mpl/test_circuit_matplotlib_drawer.py (hunk 4 above)
Comment (1ucian0) [reply test/ipynb/mpl/test_circuit_matplotlib_drawer.py]: split in 22383384cfaf1d5bc44b2241984636de997e7fe1
